cartography/intel/azure/sql.py

```diff
@@ -0,0 +1,855 @@
+
+    except HttpResponseError as e:
+        logger.warning(f"Error while retrieving servers - {e}")
+        return []
+
+    for server in server_list:
+        x = server['id'].split('/')
+        server['resourceGroup'] = x[x.index('resourceGroups') + 1]
+
+    return server_list
+
+
+@timeit
+def load_server_data(
+        neo4j_session: neo4j.Session, subscription_id: str, server_list: List[Dict],
+        azure_update_tag: int,
+) -> None:
+    """
+    Ingest the server details into neo4j.
+    """
+    ingest_server = """
+    UNWIND {server_list} as server
+    MERGE (s:AzureSQLServer{id: server.id})
+    ON CREATE SET s.firstseen = timestamp(),
+    s.resourcegroup = server.resourceGroup, s.location = server.location
+    SET s.lastupdated = {azure_update_tag},
+    s.name = server.name,
+    s.kind = server.kind,
+    s.state = server.state,
+    s.version = server.version
+    WITH s
+    MATCH (owner:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})
+    MERGE (owner)-[r:RESOURCE]->(s)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {azure_update_tag}
+    """
+
+    neo4j_session.run(
+        ingest_server,
+        server_list=server_list,
+        AZURE_SUBSCRIPTION_ID=subscription_id,
+        azure_update_tag=azure_update_tag,
+    )
+
+
+@timeit
+def sync_server_details(
+        neo4j_session: neo4j.Session, credentials: Credentials, subscription_id: str,
+        server_list: List[Dict], sync_tag: int,
+) -> None:
+    details = get_server_details(credentials, subscription_id, server_list)
+    load_server_details(neo4j_session, credentials, subscription_id, details, sync_tag)
+
+
+@timeit
+def get_server_details(
+        credentials: Credentials, subscription_id: str, server_list: List[Dict],
+) -> Generator[Any, Any, Any]:
+    """
+    Iterate over each servers to get its resource details.
+    """
+    for server in server_list:
+        dns_alias = get_dns_aliases(credentials, subscription_id, server)
+        ad_admins = get_ad_admins(credentials, subscription_id, server)
+        r_databases = get_recoverable_databases(credentials, subscription_id, server)
+        rd_databases = get_restorable_dropped_databases(credentials, subscription_id, server)
+        fgs = get_failover_groups(credentials, subscription_id, server)
+        elastic_pools = get_elastic_pools(credentials, subscription_id, server)
+        databases = get_databases(credentials, subscription_id, server)
+        yield server['id'], server['name'], server[
+            'resourceGroup'
+        ], dns_alias, ad_admins, r_databases, rd_databases, fgs, elastic_pools, databases
+
+
+@timeit
+def get_dns_aliases(credentials: Credentials, subscription_id: str, server: Dict) -> List[Dict]:
+    """
+    Returns details of the DNS aliases in a server.
+    """
+    try:
+        client = get_client(credentials, subscription_id)
+        dns_aliases = list(
+            map(
```

Comment:
Non-blocker: list comprehension

Reply:
We wanted to keep the code consistent throughout all Azure modules and since we used list(map(...)) in the compute.py code. What I'll do is I'll open a PR in the future and change all the list(map(...)) codes to list comprehension across all Azure modules we'll add. :)